### PR TITLE
Be robust to updated names and missing github field. connects 18F/team-api.18f.gov#123

### PR DESCRIPTION
**Pull request by ertzeid**
When updating to add Start and End dates to team members, it surfaced the need for team_api to be able to handle a bit more.

### CODE REVIEW

**Comment by ccostino** [comment]
@ertzeid:  The changes themselves look okay, but I want to make sure I'm understanding what's happening correctly.  This is providing support for looking team members up and cross referencing them to each other based on other last names that team members may have had?  And it does this by taking into account those older names but ultimately using the current last name for the actual cross-referencing and joining?

Plus the more robust support for any missing `github` fields.

@mbland:  The CI failures appear to be completely unrelated to this change as far as I can tell.  Specifically, they are failing on the `test_join_project` test in `test/joiner_join_project_data_test.rb` (starting on line 22).  There appears to be an extraneous `draft` key in the actual output that is missing from the test data, but I cannot reproduce this locally.  Do you have any idea what is happening here or where that extra key might be coming from?

File: lib/team_api/cross_referencer.rb [hunk 1]
```diff
@@ -41,6 +45,15 @@ def self.create_xref_data(site)
     end
     private_class_method :create_xref_data
 
+    def self.create_name_xrefs(team)
+      name_xrefs = {}
+      team.data.each do |member|
+        name_xrefs[member.last['deprecated_name']] =
+            member.last['name'] if member.last['deprecated_name']
```
Comment (ccostino) [inline lib/team_api/cross_referencer.rb]: Does this need to perform any downcasing right here?

**Comment by ccostino** [comment]
Slight correction based on the data changes in the other repo I see; it's not their last name, it's the previous username they were using, right?

File: lib/team_api/cross_referencer.rb (hunk 1 above)
Comment (ertzeid) [inline lib/team_api/cross_referencer.rb]: No, it shouldn't need to do any case correction. This is all coming from existing data, so it should already have the correct cases (which also all happen to be all-lowercase).

**Comment by ertzeid** [comment]
Yes: the 'name' field used to have names that were created over time and thus had no main pattern to them. For the start/end date, I wanted to standardize the filenames so that I (and others eventually, I'm sure) could work with them more easily. But that caused a host of cross-referencing problems, so this change mainly adds the functionality to be able to cross-reference based on both current and previous name, as different values are used in different project and workgroup `.about.yml` files.

File: lib/team_api/joiner.rb [hunk 2]
```diff
@@ -64,6 +64,7 @@ def team_members
       @team_members ||= team.map { |key, value| value unless key == 'private' }
         .compact
         .concat((team['private'] || {}).values)
+      @team_members
```
Comment (mbland) [inline lib/team_api/joiner.rb]: This method should already return `@team_members` without this explicit statement. Does it not?

**Comment by mbland** [comment]
One minor comment aside, this is looking really good. Looking into the Travis failure now.

File: lib/team_api/joiner.rb (hunk 2 above)
Comment (ertzeid) [inline lib/team_api/joiner.rb]: Oops, left over from when I was `puts`ing some stuff here. Removing...

File: lib/team_api/cross_referencer.rb (hunk 1 above)
Comment (ccostino) [inline lib/team_api/cross_referencer.rb]: Ah okay, thanks!

**Comment by mbland** [comment]
Oh, that's why: Jekyll 3. @ertzeid, perhaps add `delete doc['draft']` to `setup`?

**Comment by mbland** [comment]
Correction: `doc.data.delete('draft')`. Too much Node lately.

**Comment by ertzeid** [comment]
Done. Waiting for results.....

**Comment by ccostino** [comment]
Woohoo!

**Pull request merged**